Added possibility to draw area between 2 lines in the line chart

krispypen opened the pull request:
This resolves issue #93 

With this you can draw an area between 2 lines, the area can be a color (usually something semi-transparent), or a gradient. You can use it by adding a betweenBarsData property on LineChartData.

And also adds it to the example project:
<img width="371" alt="Screenshot 2019-12-15 at 10 45 31" src="https://user-images.githubusercontent.com/156955/70860980-6f9e0000-1f28-11ea-92d9-781a97a7a619.png">

imaNNeoFighT commented:
Hi @krispypen , great to see this, 
I appreciate it, I'm gonna review it ASAP.
Thanks!

imaNNeoFighT commented on example/lib/line_chart/samples/line_chart_sample_betweenbarsdata.dart:
```diff
@@ -0,0 +1,146 @@
+import 'package:fl_chart/fl_chart.dart';
+import 'package:flutter/material.dart';
+
+class LineChartSampleBetweenBarsData extends StatelessWidget {
```
Please follow our namings, (line_chart_sample{x}.dart), x is a number, we should follow it to prevent making inconsistencies (we use this convention for showing samples in documentations, and we name the images that have been saved in `./repo_files/images/line_chart/line_chart_sample_{x}.png`),
Please rename it to `line_chart_sample7.dart` and put it in the last sample position (we add samples sequentially)

imaNNeoFighT commented on lib/src/chart/line_chart/line_chart_painter.dart:
```diff
@@ -200,17 +215,20 @@ class LineChartPainter extends AxisChartPainter<LineChartData> with TouchHandler
   /// second one is curved corners line on spot connections,
   /// and we use isCurved to find out how we should generate it,
-  Path _generateBarPath(Size viewSize, LineChartBarData barData) {
+  Path _generateBarPath(Size viewSize, LineChartBarData barData, Path appendToPath) {
```
May I ask you to explain the usage of `appendToPath`?

krispypen replied:
I added some documentation:

`If you want to concatenate paths together for creating an area between multiple bars for example, you can pass the appendToPath`

Is that better? Maybe you know a better way to concatenate path's?

imaNNeoFighT commented on lib/src/chart/line_chart/line_chart_painter.dart:
```diff
@@ -78,6 +78,10 @@ class LineChartPainter extends AxisChartPainter<LineChartData> with TouchHandler
     }
 
+    for(BetweenBarsData betweenBarsData in data.betweenBarsData) {
+      drawBetweenBarsArea(canvas, size, data.lineBarsData[betweenBarsData.fromIndex], data.lineBarsData[betweenBarsData.toIndex], betweenBarsData);
```
We can have this code `data.lineBarsData[betweenBarsData.fromIndex]` inside the function, instead of passing it to function, because you are passing `betweenBarsData`, then we can retrieve these two arguments from it, as you know we should not pass lots of arguments to a function, it makes complication.

imaNNeoFighT commented on lib/src/chart/line_chart/line_chart_painter.dart:
```diff
@@ -139,6 +143,17 @@ class LineChartPainter extends AxisChartPainter<LineChartData> with TouchHandler
     _drawBar(canvas, viewSize, barPath, barData);
   }
 
+  void drawBetweenBarsArea(Canvas canvas, Size viewSize, LineChartBarData fromBarData, LineChartBarData toBarData, BetweenBarsData betweenBarsData) {
```
Also here, please retrieve the `fromBarData` and `toBarData` inside the function using `betweenBarsData`.

imaNNeoFighT commented on lib/src/chart/line_chart/line_chart_painter.dart:
```diff
@@ -139,6 +143,17 @@ class LineChartPainter extends AxisChartPainter<LineChartData> with TouchHandler
+    final List<FlSpot> spots = [];
+    //spots.addAll(fromBarData.spots);
+    spots.addAll(toBarData.spots.reversed.toList());
+    //final barPath = _generateBarPath(viewSize, toBarData.copyWith(spots: spots));
```
I think these comments are trash, please remove useless comments.

imaNNeoFighT commented on lib/src/chart/line_chart/line_chart_painter.dart:
```diff
@@ -200,17 +215,22 @@ class LineChartPainter extends AxisChartPainter<LineChartData> with TouchHandler
-  Path _generateBarPath(Size viewSize, LineChartBarData barData) {
+  /// If you want to concatenate paths together for creating an area between
+  /// multiple bars for example, you can pass the appendToPath
+  Path _generateBarPath(Size viewSize, LineChartBarData barData, Path appendToPath) {
```
May I ask you put the `appendToPath` inside { }, to make it optional? (default null)

imaNNeoFighT commented on lib/src/chart/line_chart/line_chart_painter.dart:
```diff
@@ -124,7 +128,7 @@ class LineChartPainter extends AxisChartPainter<LineChartData> with TouchHandler
 
   void drawBarLine(Canvas canvas, Size viewSize, LineChartBarData barData) {
-    final barPath = _generateBarPath(viewSize, barData);
+    final barPath = _generateBarPath(viewSize, barData, null);
```
https://github.com/imaNNeoFighT/fl_chart/pull/136/files#r360580571

imaNNeoFighT commented:
May I ask you to update the [LineChartData](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/line_chart.md#linechartdata) documentation?
I promise this is the last one :)

imaNNeoFighT commented:
Hi there, any news?

krispypen commented:
Hi, sorry for the delay, just implemented every request, also added screenshot of sample 7 in the docs (but the url to the image will work only after merging the PR)

imaNNeoFighT commented:
Hi man.
Thanks for your awesome work, I appreciate it.
Hope to see your more PRs.
Cheers!
